client/app/components/composites/MenuPriority/MenuPriority.js

```diff
@@ -0,0 +1,166 @@
+import { Component, PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+import r, { div, a } from 'r-dom';
+import _ from 'lodash';
+import classNames from 'classnames';
+import { canUseDOM } from '../../../utils/featureDetection';
+
+import Menu from '../Menu/Menu';
+import css from './MenuPriority.css';
+
+const EXTRA_SPACING_RIGHT = 24;
+const LINK_SPACING = 18;
+const ROUNDING_ERROR_MARGIN = 2;
+
+class MenuPriority extends Component {
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.links = _.sortBy(this.props.content, ['priority']);
+    this.componentDidMount = this.componentDidMount.bind(this);
+    this.updateWidths = this.updateWidths.bind(this);
+    this.updateNav = this.updateNav.bind(this);
+
+    this.state = {
+      priorityWrapperWidth: '0px',
+      priorityLinks: this.links.filter((l) => l.priority >= 0),
+      hiddenLinks: this.links,
+    };
+  }
+
+  componentDidMount() {
+    if (canUseDOM) {
+      const that = this;
```

Comment:
`this` within an arrow function refers to the same object as the `this` in the outer scope, so this is unnecessary as `that` is only used within arrow functions.